FileShares/src/FileSharesWeb/FileShareHealthContributor.cs

```diff
@@ -0,0 +1,62 @@
+using System.Runtime.InteropServices;
+using Steeltoe.Common.HealthChecks;
+
+namespace Steeltoe.Samples.FileSharesWeb;
+
+internal class FileShareHealthContributor : IHealthContributor
+{
+    private const ulong ThresholdInBytes = 10 * 1024 * 1024;
+    public string Id => "fileShareHealthContributor";
+
+    public Task<HealthCheckResult?> CheckHealthAsync(CancellationToken cancellationToken)
+    {
+        HealthCheckResult result = Health();
+        return Task.FromResult<HealthCheckResult?>(result);
+    }
+
+    private static HealthCheckResult Health()
+    {
+        if (!string.IsNullOrEmpty(FileShareHostedService.Location) && Directory.Exists(FileShareHostedService.Location))
+        {
+            if (NativeMethods.GetDiskFreeSpaceEx(FileShareHostedService.Location, out ulong freeBytesAvailable, out ulong totalNumberOfBytes,
+                out ulong totalNumberOfFreeBytes))
+            {
+                var result = new HealthCheckResult
+                {
+                    Status = totalNumberOfFreeBytes >= ThresholdInBytes ? HealthStatus.Up : HealthStatus.Down
+                };
+
+                result.Details.Add("totalBytes", totalNumberOfBytes);
+                result.Details.Add("freeBytes", totalNumberOfFreeBytes);
+                result.Details.Add("freeBytesAvailable", freeBytesAvailable);
+                result.Details.Add("minimumFreeBytes", ThresholdInBytes);
```

Comment:
I don't find these names very understandable.

Reply:
Better?
```
result.Details.Add("bytesFreeForUser", freeBytesAvailable);
result.Details.Add("totalFreeBytes", totalNumberOfFreeBytes);
result.Details.Add("minimumFreeBytes", ThresholdInBytes);
result.Details.Add("totalCapacityBytes", totalNumberOfBytes);
result.Details.Add("path", FileShareHostedService.Location);
```

Reply:
Yes, but it doesn't align with what the Steeltoe health contributor returns. I don't think `totalCapacityBytes` should be reported at all. Furthermore, the Steeltoe contributor reports `exists`.

Reply:
Why shouldn't `totalCapacityBytes` be reported? I could see dropping `minimumFreeBytes` since it's hardcoded.

Reply:
Because its value is irrelevant for the disk space status. An app can write until its quota threshold is exceeded. The physical space is an implementation detail of the file share server.

Reply:
The space check is wrong, it should take the quota into account.

Reply:
> The space check is wrong, it should take the quota into account.

Do you know how we can get quota information? I'm not seeing much online and I've not done that before

Reply:
It's the value of the `lpFreeBytesAvailableToCaller` parameter, see https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getdiskfreespaceexa#parameters.

Reply:
This was handled in the Steeltoe PR